documentation/src/viewerUtilities.ts

```diff
@@ -0,0 +1,21 @@
+import { Cognite3DViewer } from '@cognite/reveal';
+import { CameraChangeDelegate, PointerEventDelegate } from '@cognite/reveal/public/migration/Cognite3DViewer';
+
+type InternalEventListeners = {
+  cameraChange: CameraChangeDelegate[];
+  click: PointerEventDelegate[];
+  hover: PointerEventDelegate[];
+}
+
+/**
+ * Resets all event handlers on the viewer provided. This uses
+ * internal functionality and actual applications needs to implement
+ * this to better approaches than this.
+ */
+export function resetViewerEventHandlers(viewer: Cognite3DViewer): void {
+  // @ts-expect-error
+  const eventListeners: InternalEventListeners = viewer.eventListeners;
```

Comment:
Still don't understand, sorry :) My point was mainly against having `@ts-expect-error`, I just don't understand what's going wrong with the types here and why. `viewer.eventListeners` should already have exactly the same type. So what goes wrong if you remove InternalEventListeners from here?

Reply:
Oh, I see, eventListeners is private thing. Wouldn't it be better then just make it public readonly or add a method to reset all listeners? Like... `viewer.offEverything` or something that just sounds better but does what you're trying to do here with private things?

Reply:
That for sure could be done later since docs are using published npm version instead of the one from master.

Reply:
It felt a bit horrible yes. I don't think we really should expose eventListeners, but it might be an option to have a function to clear all listeners. It's a bit smelly as well, but might be better than this.